src/indicator-fixes.js

```diff
@@ -0,0 +1,249 @@
+
+
+function fieldToString(f) {
+  if (!f.subfields) {
+    return `${f.tag}    ${f.value}`;
+  }
+  return `${f.tag} ${f.ind1}${f.ind2} ‡${formatSubfields(f)}`;
+
+  function formatSubfields(field) {
+    //return field.subfields.map(sf => `${sf.code}${sf.value || ''}`).join('‡');
+    return field.subfields.map(sf => `${sf.code}${sf.value}`).join(' ‡');
+  }
+}
+
+/*
+function nvdebug(message, func) {
+  if (func) { // eslint-disable-line functional/no-conditional-statement
+    func(message);
+  }
+  console.info(message); // eslint-disable-line no-console
+}
+*/
+
+export default function () {
+
+  // NB! We should and could handle ISNIs here as well.
+  return {
+    description: 'Normalizes indicator values',
+    validate, fix
+  };
+
+  function fix(record) {
+    const res = {message: [], fix: [], valid: true};
+    //message.fix = []; // eslint-disable-line functional/immutable-data
+
+    // Actual parsing of all fields
+    /*
+      if (!record.fields) {
+        return false;
+      }
+      */
+
+
+    recordNormalizeIndicators(record);
+
+
+    // message.valid = !(message.message.length >= 1); // eslint-disable-line functional/immutable-data
+    return res;
+  }
+
+  function validate(record) {
+    const res = {message: []};
+    //nvdebug(`NORMALIZE CONTROL NUMBER VALIDATE`, debug);
+    // Actual parsing of all fields
+    /*
+      if (!record.fields) {
+        return false;
+      }
+      */
+
+    validateRecord(record, res);
+
+    res.valid = res.message.length < 1; // eslint-disable-line functional/immutable-data
+    return res;
+  }
+
+
+  function validateRecord(record, res) {
+    //nvdebug(record);
+    const clonedFields = JSON.parse(JSON.stringify(record.fields));
+    recordNormalizeIndicators(record);
+
+    record.fields.forEach((field, index) => compareFields(field, index));
+
+    function compareFields(field, index) {
+      const origFieldAsString = fieldToString(clonedFields[index]);
+      //const clonedFieldAsString = fieldToString(field);
+      if (clonedFields[index].ind1 !== field.ind1) { // eslint-disable-line functional/no-conditional-statement
+        //nvdebug(`FIX IND1: '${clonedFields[index].ind1}' => '${field.ind1}': ${clonedFieldAsString}`);
+        res.message.push(`Expected IND1 for '${origFieldAsString}' is '${field.ind1}'`); // eslint-disable-line functional/immutable-data
+      }
+      if (clonedFields[index].ind2 !== field.ind2) { // eslint-disable-line functional/no-conditional-statement
+        //nvdebug(`FIX IND2: '${clonedFields[index].ind2}' => '${field.ind2}': ${clonedFieldAsString}`);
+        res.message.push(`Expected IND2 for '${origFieldAsString}' is '${field.ind2}'`); // eslint-disable-line functional/immutable-data
+      }
+    }
+    // Validator should not change the original record:
+    record.fields = clonedFields; // eslint-disable-line functional/immutable-data
```

Comment:
We could avoid needing to update the record by cloning the record to recordForValidation (or something like that) in the beginning.